proto-public/pbmesh/v1alpha1/pbproxystate/traffic_permissions.proto

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v1alpha1.pbproxystate;
+
+message L7TrafficPermission {}
+
+enum TrafficPermissionAction {
+  INTENTION_ACTION_DENY = 0;
+  INTENTION_ACTION_ALLOW = 1;
+}
+
+message L4TrafficPermissions {
+  TrafficPermissionAction default_action = 1;
+  // permission is constructed by pre-flattening all relevant permissions.
+  repeated L4Permission permissions = 2;
```

Comment:
It seems like with this you could end up with a list of lists of allow and deny principals which will result in many RBAC filters (i.e. more than 2) unless we flatten it into two? I was imagining that we would combine all allow policies into one rbac filter and all deny policies into one rbac which will then mean that we will always have at most two rbac filters for any filter chain. This would then result in in-lining this object here:

```
message L4TrafficPermissions {
    repeated L4Principal allow_principals = 1;
    repeated L4Principal deny_principals = 2;
}
```

This would then be similar to how computed traffic permissions are structured also.

Reply:
Yeah, it was a bit awkward, but ended up getting flattened into two or fewer RBAC filters. The solution we came up with earlier fixes the awkwardness.